src/System.Private.ServiceModel/src/System/ServiceModel/Channels/RequestChannel.cs

```diff
@@ -196,10 +197,15 @@ private void ReleaseRequest(IRequestBase request)
                 _outstandingRequests.Remove(request);
                 if (_outstandingRequests.Count == 0)
                 {
-                    if (!_closed && _closedTcs != null)
+                    // When we are closed or closing, _closedTcs is managed by the close logic.
+                    if (!_closed && State != CommunicationState.Closing)
                     {
-                        _closedTcs.TrySetResult(null);
-                        _closedTcs = null;
+                        // Protect against close altering _closedTcs concurrently
+                        var closedTcs = Interlocked.CompareExchange(ref _closedTcs, null, _closedTcs);
+                        if (closedTcs != null)
+                        {
+                            closedTcs.TrySetResult(null);
```

Comment:
Question only: there is nothing adverse happening if tcs.TrySetResult fails, right? My read is that this returns false if the tcs is already in a final state, so just wondering if this meant something weird could have happened